test/integration/targets/tower_job_template/tasks/main.yml

```diff
@@ -52,3 +52,19 @@
 - assert:
     that:
       - "result is changed"
+
+- name: Create a Job Template with extra_vars
+  tower_job_template:
+    name: hello-world
+    project: Job Template Test Project
+    inventory: Demo Inventory
+    playbook: hello_world.yml
+    machine_credential: Demo Credential
+    job_type: run
+    extra_vars: test=value
```

Comment:
I didn't think that the server supports this format of extra_vars. When I try to PATCH with these extra_vars I get

```json
{
    "extra_vars": [
        "Cannot parse as JSON (error: No JSON object could be decoded) or YAML (error: Input type `str` is not a dictionary)."
    ]
}
```

The server wants YAML or JSON, not key=value syntax, do you know if Ansible, itself, is converting it?

I'm not sure what values are actually being passed to the server, and how this test manages to pass. This kind of makes me question that some hidden bug lurks somewhere. It may still be possible to debug this with the assertion below, and checking the extra_vars returned from the server might give us a hint as to what happened.

Reply:
I can not reproduce this error. Here is my output and package versions:
```
$ ansible --version
ansible 2.4.2.0
python version = 2.7.14 (default, Mar 22 2018, 14:43:05) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```
Output of ansible-playbook test.yml -vvv:
```
changed: [localhost] => {
    "changed": true,
    "id": 19,
    "invocation": {
        "module_args": {
            "ask_credential": false,
            "ask_extra_vars": false,
            "ask_inventory": false,
            "ask_job_type": true,
            "ask_limit": false,
            "ask_tags": false,
            "become_enabled": false,
            "cloud_credential": null,
            "description": null,
            "extra_vars": "test=value",
            "extra_vars_path": null,
            "forks": null,
            "host_config_key": null,
            "inventory": "Test inventory",
            "job_tags": "handler",
            "job_type": "check",
            "limit": "handlers",
            "machine_credential": "SSH Key Rob (Machine)",
            "name": "Test template",
            "network_credential": null,
            "playbook": "some_config.yml",
            "project": "Example project",
            "skip_tags": null,
            "state": "present",
            "tower_config_file": null,
            "tower_host": "https://awx.example.com/",
            "tower_password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "tower_username": "admin",
            "tower_verify_ssl": true,
            "verbosity": null
        }
    },
    "job_template": "Test template",
    "state": "present"
}
```

Output of 'pip list | grep ansible':
```
ansible (2.4.2.0)
ansible-tower-cli (3.2.1)
```
Output snippet of 'tower-cli job_template create --help':
```
 -e, --extra-vars VARIABLES      Extra variables used by Ansible in YAML or
                                  key=value format. Use @ to get YAML from a
                                  file.
```
And the latest version of AWX

Reply:
I have tested these variations of passing extra_vars and they are always passed on as str:
```
extra_vars: {'test': 'value1', 'some': 'value2'}
extra_vars: "{'test': 'value1', 'some': 'value2'}"
extra_vars: test=value1 some=value2
```
The type of module.params.get('extra_vars') for these lines is always 'str'.

Reply:
If your server accepted the 3rd format, then it's probably reflecting a bug in Tower/AWX. If you put that data into a job template and then run a job, I suspect that the vars will not be passed into the Ansible CLI correctly, because the code to _load_ that data is not equipped to handle the key=value syntax. Maybe old versions of the Tower API accepted the syntax when modifying a JT, but they should not have.

Reply:
@AlanCoding According to the tower-cli client man the key=value syntax is supported. And this converts in a tower job template to: 'key: value'
```
$ tower-cli job_template create --help
```